### PR TITLE
User policy tests #4

### PR DESCRIPTION
**Pull request by armsasmart**
unit test in UserPolicy #4 
- restoreUser
- changePassword

### CODE REVIEW

**Comment by Tjoosten** [comment]
Hi @armsasmart! 

This looks neat on a first sight. Will try the unit tests here locally to make sure they pass. And if they pass i will merge the PR. 

Thanks x1000 For this PR.

**Comment by armsasmart** [comment]
If I have an error please advise me. I am very excited to contribute to open-source.

File: app/Domains/Users/Http/Controllers/RestoreController.php [hunk 1]
```diff
@@ -34,11 +35,12 @@ public function __construct()
      *
      * @throws \Illuminate\Auth\Access\AuthorizationException
      */
-    public function __invoke($user, DeleteAction $userDeleteAction): RedirectResponse
+    public function __invoke(int $user, DeleteAction $userDeleteAction): RedirectResponse
```
Comment (Tjoosten) [inline app/Domains/Users/Http/Controllers/RestoreController.php:38]: Nice catch!

File: app/Domains/Users/Policies/UserPolicy.php [hunk 2]
```diff
@@ -14,7 +14,7 @@ class UserPolicy
     /**
      * Determine whether the authenticated user can delete an other user or not.
      *
-     * @param  User $user The resource entity from the authenticated user.
+     * @param User $user The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:17]: Hey Can u align the User class with the same start as the return type in the docblock.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 3]
```diff
@@ -25,7 +25,7 @@ public function delete(User $user): bool
     /**
      * Determine whether the authenticated user can view other users or not.
      *
-     * @param  User $user The resource entity from the authenticated user.
+     * @param User $user The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:28]: Hey Can u align the User class with the same start as the return type in the docblock.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 4]
```diff
@@ -36,7 +36,7 @@ public function view(User $user): bool
     /**
      * Determine whether the user can create new users in the application or not.
      *
-     * @param  User $user  The resource entity from the authenticated user.
+     * @param User $user The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:39]: Hey Can u align the User class with the same start as the return type in the docblock.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 5]
```diff
@@ -47,7 +47,7 @@ public function create(User $user): bool
     /**
      * Determine whether the authenticated user can update other users or not.
      *
-     * @param  User $user  The resource entity from the authenticated user.
+     * @param User $user The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:50]: Hey Can u align the User class with the same start as the return type in the docblock.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 6]
```diff
@@ -58,8 +58,8 @@ public function update(User $user): bool
     /**
      * Determine whether the authenticated user can lock other user accounts.
      *
-     * @param  User $user  The resource entity from the authenticated user.
-     * @param  User $model The resource entity from the given user.
+     * @param User $user The resource entity from the authenticated user.
+     * @param User $model The resource entity from the given user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:62]: Hey,

Can u align the User class with the same start as the return type in the docblock.
Also the start of the description text.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 7]
```diff
@@ -71,8 +71,8 @@ public function lock(User $user, User $model): bool
     /**
      * Determine whether the authenticated user can unlock an user account or not.
      *
-     * @param  User $user  The resource entity from the authenticated user.
-     * @param  User $model The resource entity from the given user.
+     * @param User $user The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:74]: Hey,

Can u align the User class with the same start as the return type in the docblock.
Also the start of the description text.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 8]
```diff
@@ -85,21 +85,21 @@ public function unlock(User $user, User $model): bool
     /**
      * Determine whether the authenticated user can restore a 'deleted' user account or not.
      *
-     * @param  User $user  The resource entity from the authenticated user.
-     * @param  User $model The resource entity from the given user.
+     * @param User $user The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:88]: Hey,

Can u align the User class with the same start as the return type in the docblock.
Also the start of the description text.

File: app/Domains/Users/Policies/UserPolicy.php (hunk 2 above)
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:17]: Hey, 

Can u align the docblock with one extra space. 

Example: 

**now:** 

```
/**
  * Determine whether the authenticated user can delete an other user or not.
  *
  * @param  User $user The resource entity from the authenticated user.
  * @param User $user The resource entity from the authenticated user.
  * @return bool
  */
```

**what i mean:** 

```
/**
  * Determine whether the authenticated user can delete an other user or not.
  *
  * @param  User $user The resource entity from the authenticated user.
  * @param  User $user The resource entity from the authenticated user.
  * @return bool
  */
```

File: app/Domains/Users/Policies/UserPolicy.php (hunk 3 above)
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:28]: The docblock alignment seems a bit off.

File: app/Domains/Users/Policies/UserPolicy.php (hunk 4 above)
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:39]: The docblock alignment seems a bit off.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 9]
```diff
@@ -58,8 +58,8 @@ public function update(User $user): bool
     /**
      * Determine whether the authenticated user can lock other user accounts.
      *
-     * @param  User $user  The resource entity from the authenticated user.
-     * @param  User $model The resource entity from the given user.
+     * @param User $user The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:61]: The docblock alignment seems a bit off.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 10]
```diff
@@ -47,7 +47,7 @@ public function create(User $user): bool
     /**
      * Determine whether the authenticated user can update other users or not.
      *
-     * @param  User $user  The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:50]: The docblock alignment seems a bit off.

File: app/Domains/Users/Policies/UserPolicy.php (hunk 5 above)
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:50]: The docblock alignment seems a bit off.

File: app/Domains/Users/Policies/UserPolicy.php (hunk 8 above)
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:88]: The docblock alignment seems a bit off.

File: app/Domains/Users/Policies/UserPolicy.php [hunk 11]
```diff
@@ -85,21 +85,21 @@ public function unlock(User $user, User $model): bool
...
             && $user->hasKioskUserGroup()
-            || $user->hasKioskUserGroup() && $user->hasPermissionTo('restore-users');
+            && $user->hasPermissionTo('restore-users');
     }
 
     /**
      * Determine whether the user can change passwords from other users or not.
      *
-     * @param  User $user The resource entity from the authenticated user.
+     * @param User $user The resource entity from the authenticated user.
```
Comment (Tjoosten) [inline app/Domains/Users/Policies/UserPolicy.php:102]: The docblock alignment seems a bit off.

**Review by Tjoosten — CHANGES_REQUESTED** [review]
Hey, 

I found some nitpicks in the code. The docblocks seems a bit off. But i prefer the params aligned with the returns. Because imo it is better for readability.

**Comment by armsasmart** [comment]
Oh, My ide auto format. I'm done. 🎉

**Review by Tjoosten — APPROVED** [review]

**Pull request merged**